### PR TITLE
Switch to UniFFI library mode [ci full]

### PR DESCRIPTION
**Pull request by bendk**
I'm reviving #5592 and giving it another shot.  If successful, this can be the basis for @gruberb's proc-macro work.

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [x] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due diligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by bendk** [comment]
In that PR, Tarik said "Looks like this ended up being a breaking change for firefox-ios because of a swift name conflict".  I completely forget what that means, but can someone test this code with a local ios build and see if it works?

**Review by gruberb — APPROVED** [review]
I ran the PR locally against an firefox-ios and it worked. So I approve, but not sure if we want another pair of eyes.

**Comment by bendk** [comment]
I tested a local Fenix build and it worked for me.  @gruberb was able to get iOS to successfully compile.  This seems like good enough evidence to me, but please tell me if we should test more.

@jeddai do you think we need to do any extra testing for cirrus?

File: components/autofill/uniffi.toml
Comment (gruberb) [inline components/autofill/uniffi.toml:3]: I am curious: Why can we remove the `cdylib_name`?

Comment (bendk) [reply components/autofill/uniffi.toml:3]: That's one perk of library mode, since we're passing the library path to the bindings generator we can calculate the cdylib name from that.

File: docs/howtos/adding-a-new-component.md
Comment (bendk) [inline docs/howtos/adding-a-new-component.md:162]: I tried to update this as best I could, but I was unsure of some Swift things.
- Do my current changes make sense?
- What should we do with the above line?  In a proc-macro world, we won't have a UDL file to add to compile sources.  Do we need to add some other file?  Or is this not needed anymore?

Comment (linabutler) [reply docs/howtos/adding-a-new-component.md:162]: I think your current changes make sense!

For the above line...I'm not sure just yet! How do we invoke `uniffi-bindgen` to generate the Swift bindings, and what does it take for the input, in the proc-macro world?

The Build Rule that we have now takes all `.*udl` files as inputs, and runs `uniffi-bindgen generate` on each one; but we can change it to whatever it needs!

Comment (bendk) [reply docs/howtos/adding-a-new-component.md:162]: The new command is `cargo uniffi-bindgen generate --library [path-to-megazord-ios-a] -l swift -o [out-dir]`.  The input is the megazord library, which is awkward to specify in the compile sources.  Can we add a build step that unconditionally runs that after the megazord is built?  Or is my change to `build-xcframework.sh` already enough?

File: megazords/ios-rust/build-xcframework.sh
Comment (linabutler) [inline megazords/ios-rust/build-xcframework.sh:142]: "Arbitrary target" here means that it doesn't matter whether we use `aarch64-apple-ios`, `aarch64-apple-ios-sim`, or `x86_64-apple-ios` (because we build all three, and the metadata for all three should be identical), right?

File: publish.gradle
Comment (linabutler) [inline publish.gradle:228]: `t.get()` (and `tasks[taskName]`) will force evaluation of the task graph—which can have unintended consequences down the line—and there's new lazy configuration APIs that we can use here, for both Gradle and the Android Gradle plug-in. Not at all a blocker for this PR, since we were already doing this, but I can get a follow-up bug on file for that!

**Review by linabutler — APPROVED** [review]
This looks terrific, @bendk! :shipit:

File: megazords/ios-rust/build-xcframework.sh
Comment (bendk) [reply megazords/ios-rust/build-xcframework.sh:142]: Yes that's exactly right.  In theory, with proc-macros, you could have different definitions depending on the target, but we don't do that yet.

**Pull request merged**
